TcgPocket/App/src/pages/settings/modules/PasswordForm.tsx

```diff
@@ -0,0 +1,122 @@
+import { useForm } from '@mantine/form';
+import { useFormValidation } from '../../../helpers/useFormValidation';
+import {
+  SignInUserDto,
+  UserGetDto,
+  UserPasswordUpdateDto,
+} from '../../../types/users';
+import { Flex, PasswordInput, createStyles } from '@mantine/core';
+import { SecondaryButton } from '../../../components/buttons/SecondaryButton';
+import { PrimaryButton } from '../../../components/buttons/PrimaryButton';
+import { dispatch } from '../../../store/configureStore';
+import { updateUserPassword } from '../../../services/AuthServices';
+import { error, success } from '../../../services/notification';
+
+type PasswordFormProps = {
+  user: UserGetDto;
+  loginUserAfterUpdate: (arg: SignInUserDto) => void;
+};
+
+type PasswordForm = Omit<UserPasswordUpdateDto, 'userName'>;
+
+const initialValues = {
+  currentPassword: '',
+  newPassword: '',
+  newPasswordConfirmation: '',
+} as const;
+
+export function PasswordForm({
+  user,
+  loginUserAfterUpdate,
+}: PasswordFormProps): React.ReactElement {
+  const { classes } = useStyles();
+  const { validatePassword } = useFormValidation();
+
+  const form = useForm({
+    initialValues: initialValues,
+    validateInputOnBlur: true,
+    validate: {
+      currentPassword: (value) => (value === '' ? 'Cannot be Empty' : null),
+      newPassword: (value) =>
+        validatePassword(value) ? 'Invalid Password' : null,
+      newPasswordConfirmation: (value, values) =>
+        value !== values.newPassword ? 'Passwords do not match' : null,
+    },
+  });
+
+  const handleSubmit = async (values: PasswordForm) => {
+    const userWithUpdatedPassword: UserPasswordUpdateDto = {
+      userName: user.userName,
+      currentPassword: values.currentPassword,
+      newPassword: values.newPassword,
+      newPasswordConfirmation: values.newPasswordConfirmation,
+    };
+
+    const { payload } = await dispatch(
+      updateUserPassword(userWithUpdatedPassword)
+    );
+
+    if (!payload) {
+      return;
+    } else if (payload.hasErrors) {
+      payload.errors.forEach((err) => error(err.message));
+    } else {
+      success('Password Updated');
+      form.reset();
+
+      loginUserAfterUpdate({
+        userName: user.userName,
+        password: values.newPassword,
+      });
+    }
+  };
+
+  const handleReset = () => {
+    form.reset();
+  };
+
+  const determineDisabled =
+    form.values.currentPassword === '' ||
+    form.values.newPassword === '' ||
+    form.values.newPasswordConfirmation === '';
```

Comment:
```suggestion
  const determineDisabled = !form.isValid();
```

This should make sure the button is not enabled until everything in the form complies to the validation rules.

Reply:
I remember Brandon saying something about this causing issues ? I'm not too sure so I'll leave the comment for now until we are sure :)

Reply:
If there are no empty fields, you should let the user push the button. Then and only then if the form is still invalid, display the form errors from the frontend.

Reply:
To make sure I'm understanding correctly, even if a user types in an email in the wrong format, they should still be able to click submit, at which time the errors appear?

Reply:
If you cannot press the button and it's for some obscure reason then the user will be confused and frustrated. Ideally any given field should be validated onBlur, but on submit before actually talking to the api is sufficient.